packages/ember-glimmer/lib/utils/process-args.js

```diff
@@ -13,23 +14,26 @@ export function processComponentArgs(namedArgs) {
   let props = Object.create(null);
   let args = Object.create(null);
 
-  props[ARGS] = args;
-
+  let seen = {};
   for (let i = 0; i < keys.length; i++) {
     let name = keys[i];
+
+    assert(`Component attribute \`${name}\` is declared multiple times`, !seen[name]);
+    if (seen[name]) { continue; }
+    seen[name] = true;
+
     let ref = namedArgs.get(name);
     let value = attrs[name];
 
-    if (typeof value === 'function' && value[ACTION]) {
-      attrs[name] = value;
-    } else if (ref[UPDATE]) {
+    if (!(typeof value === 'function' && value[ACTION]) && ref[UPDATE]) {
```

Comment:
If seen this check `typeof value === 'function' in multiple parts of the codebase, curious if there is a "helper" we can import to use?

Reply:
I like this cleanup 👏

Reply:
@alvincrespo inline `typeof value === 'function` is more optimizable by v8 than imported `isFunction` thats why you see it everywhere in codebase, most of things done inline

Reply:
Oh interesting, I did not know that. Thanks for the details here!